modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java

```diff
@@ -99,16 +99,9 @@ private EntityField _getEntityField(ObjectField objectField) {
 			return null;
 		}
 
-		if (objectField.compareBusinessType(
```

Comment:
What is the purpose of the `compareBusinessType` method? :thinking: is it not just a normal string comparison?

Reply:
Maybe that method should be used always to compare business types then

Reply:
If we use that method, then the validation with a List would be like:

```
if (ListUtil.exists(_unsupportedBusinessTypes, objectField::compareBusinessType)) {
	return null;
}
```

Reply:
Mmmmm, at my first look I didn't like the `compareBusinessType` method as it is the result of an equal and I didn't like and it forces me to drag the objectField every time.

Rethinking this after your comment maybe makes more sense your way and we will save some code. Sometimes I overthink and I forget about the [YAGNI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) 😓

Reply:
Ok, now I see that this is for the case that we use a Set instead of an enum. For now, I do not want to change the enum because of the reason I typed in the other comment. We can discuss it and then see if we can change it.

I tried to do the same with the enum but the resultant if clause is too ugly and I prefer to keep it in a separate method that with a name that tell us what it is doing 😁